### PR TITLE
Variable support

### PR DESCRIPTION
**Pull request by umutuluer**
Hi;

If you define a variable name and value in variables section in your yaml file, you can use this variable with prefix ${VARIABLE_NAME} etc. 

You can find example usage in spec/conf/variables.yaml

Tests are passed! But unfortunately, you can use only single variable for each command for now. I will work about this.

### CODE REVIEW

File: src/models/job.cr [hunk 1]
```diff
@@ -5,15 +5,39 @@ module Mssh
       commands: Array(String),
     )
 
-    def queued(groups : Groups)
+    def process_commands(variables : Variables?)
+      @commands.each_with_index do |cmd, i|
```
Comment (tbrand) [inline src/models/job.cr]: These implementations are little bit unstable. How about using regex? You could simply write the below by `gsub`.
Refer https://crystal-lang.org/api/0.27.2/String.html#gsub%28pattern%3ARegex%2Chash%3AHash%28String%2CUNDERSCORE%29%7CNamedTuple%29-instance-method

**Review by tbrand — COMMENTED** [review]
This is quite good! :+1:

However, I would like you to change the implementation a bit.
Could you check my comment above?

File: src/models/job.cr (hunk 1 above)
Comment (tbrand) [reply src/models/job.cr]: For example
```crystal
variables = [
  {name: "A", value: "AA"},
  {name: "B", value: "BB"}
]

commands = [
  "echo ${A} && echo ${B}",
  "echo ${A}"
]

commands.each_with_index do |cmd, i|
  commands[i] = variables.reduce(cmd) { |c, v| c.gsub(/\${#{v[:name]}}/, v[:value]) }
end

puts commands
# => ["echo AA && echo BB", "echo AA"]
```

Comment (tbrand) [reply src/models/job.cr]: Sample code with nullable variables
```crystal
#
# Replace `Array(NamedTuple(name: String, value: String))` into `Variables?`!
#
def process_commands(variables : Array(NamedTuple(name: String, value: String))?)
  commands = [
    "echo ${A} && echo ${B}",
    "echo ${A}"
  ]

  return commands unless variables_not_nil = variables

  commands.each_with_index do |cmd, i|
    commands[i] = variables_not_nil.reduce(cmd) { |c, v| c.gsub(/\${#{v[:name]}}/, v[:value]) }
  end

  puts commands
end

process_commands([{name: "A", value: "AA"}, {name: "B", value: "BB"}])
process_commands(nil)
```

Comment (umutuluer) [reply src/models/job.cr]: Thanks for explanation. It's so cool 👌

**Review by tbrand — APPROVED** [review]
Nice! I'll merge this! :+1:

As an additional thing, I think we don't need this since `Variable` has getters.
https://github.com/tbrand/mssh/pull/13/files#diff-faaef11ceda410add485d59f0e65b001R9

See the doc. https://crystal-lang.org/api/0.28.0/YAML.html
> This macro by default defines getters and setters for each variable (this can be overrided with setter and getter).

**Pull request merged**